banking/src/main/java/rocks/kata/AccountOperation.java

```diff
@@ -0,0 +1,37 @@
+
+package rocks.kata;
+
+public class AccountOperation {
+
+    private OperationType typeOfOperation;
+    private int valueOfOperation;
+    private int accountBalanceAfterOperation;
+    private CurrentDateProvider dateProvider = new CurrentDateProvider();
+    private String dateOfOperation;
+
+    public AccountOperation(OperationType typeOfOperation, int valueOfOperation, int accountBalanceAfterOperation) {
+        this.typeOfOperation = typeOfOperation;
+        this.valueOfOperation = valueOfOperation;
+        this.accountBalanceAfterOperation = accountBalanceAfterOperation;
+        this.dateOfOperation = dateProvider.getCurrentDate();
+    }
+
+    @Override
+    public String toString() {
+        return createFormattedStringForAccountOperation();
+    }
+
+    private String createFormattedStringForAccountOperation() {
```

Comment:
wygląda jak coś, co powinno być kilkoma klasami (strategie), ale może też być zaimplementowane jako enum implementujący interfejs (chociaż może to być mało elastyczne :> )

Reply:
A tu potrzebuję dokładniejszej wskazówki :)

Reply:
To jest połączone z komentarzem odnośnie enuma :p Interfejs implementowany przez enum i wartości z konkretnymi implementacjami strategii.

Reply:
To muszę doczytać ocb, bo strategie to dla mnie nowość :< Ale wrócę do tego

Reply:
Jakbyś miała wątpliwości, to pytaj, na razie nie chcę Ci podawać rozwiązania na tacy, żebyś sobie pogrzebała :p

Reply:
Coś tam wypchnęłam, do omówienia jak blisko lub daleko jest to tego co miałeś na myśli :D Doczytałam się też, że strategie przy enumach się zazwyczaj implementuje przez klasę abstrakcyjną, stąd brak interfejsu, ale ofc jestem otwarta na komentarze

Reply:
*metodę abstrakcyjną